### DISCUSSION TITLE
saving models with SDXL+LoRA

### DISCUSSION THREAD

**Discussion by taoxinhao13** [huggingface/diffusers#7452] [Q&A] [answered]
I have a question regarding saving models with SDXL+LoRA. I can successfully train and use the provided script to load LoRA weights with load_lora_weights(), but how can I save the new pipeline? I would like to treat this as a whole so that I can train the ControlNet module subsequently. I found that using pipe.save_pretrained() doesn't work, as it causes an error saying the UNet is missing many modules when loading the saved model. Could you advise on how to properly save the entire pipeline including the LoRA modifications?

**Comment by tolgacangoz** [comment, marked as answer]
Hi @taoxinhao13,
Could [fusing lora](https://huggingface.co/docs/diffusers/main/en/using-diffusers/merge_loras#fuselora) be what you are looking for?

**Reply by a-r-r-o-w** [reply]
This is it, yes! I got beat by 3 minutes :)

**Comment by a-r-r-o-w** [comment]
Hey, you could use the .fuse_lora() method to combine the unet/text-encoder parameters and their corresponding loras into one. When you save your model after that, they are saved with the lora weights as you expect.

Relevant documentation:
- https://huggingface.co/docs/diffusers/main/en/using-diffusers/merge_loras#fuselora
- https://huggingface.co/docs/diffusers/v0.23.1/en/api/loaders#diffusers.loaders.LoraLoaderMixin.fuse_lora

**Comment by taoxinhao13** [comment]
That's it!!!! Thanks a lot :)

**Reply by a-r-r-o-w** [reply]
I think you've marked my unrelated comment as the answer instead of standardAI's by mistake :) Good luck with your training 🤞

**Reply by tolgacangoz** [reply]
@taoxinhao13, I am glad you received the proper answer. But, it looks like you chose the wrong reply as the answer by mistake. Could you choose mine or Aryan's reply having the correct answer?

**Comment by tolgacangoz** [comment]
@sayakpaul It seems that the answer was not selected correctly. What can be done here?

**Reply by sayakpaul** [reply]
I think now it has been?

https://github.com/huggingface/diffusers/discussions/7452#discussioncomment-8892728
